### PR TITLE
Fix reload model in python

### LINKED ISSUE #2362: Python Model.load will return a Layer
Python Model.load / Model.load_caffe and some other methods will always return a Layer, which doesn't make sense for a Sequential or Graph.
#2168 also points out this issue. cc @qiuxin2012

### PR DESCRIPTION
**Pull request by zhichao-li**
## What changes were proposed in this pull request?
- This code
```
        model_loaded = Model.load(tmp_path)
        model_loaded.layers 
```
would fail as model_loaded is treated as `Layer` instead of `Model` or other subclass.

Fix #2362 #2168

- Fix exception in: #2388 
```
bmodel = with_bigdl_backend(kmodel)
bmodel.predict(xxx)
```
## How was this patch tested?

unittest

### CODE REVIEW

**Comment by zhichao-li** [comment]
jenkins passed http://172.168.2.101:8080/job/BigDL-PR-Validation-v3/372/

File: pyspark/bigdl/nn/layer.py [hunk 1]
```diff
@@ -1060,8 +1081,19 @@ class Sequential(Container):
 
     '''
 
-    def __init__(self, bigdl_type="float"):
-        super(Sequential, self).__init__(None, bigdl_type)
+    def __init__(self, jvalue=None, bigdl_type="float"):
+        super(Sequential, self).__init__(jvalue, bigdl_type)
+
+    @staticmethod
+    def from_jvalue(jvalue, bigdl_type="float"):
```
Comment (hkvision) [inline pyspark/bigdl/nn/layer.py]: Why there are two `from_jvalue`?

File: pyspark/bigdl/nn/layer.py [hunk 2]
```diff
@@ -115,14 +115,35 @@ def __call__(self, x=None):
...
         :return: A Python Layer
         """
-        model = Layer(jvalue, bigdl_type)
+        realClassName = callBigDlFunc(bigdl_type,
+                      "getRealClassNameOfJValue",
+                      jvalue)
+        if realClassName == "StaticGraph" or realClassName == "DynamicGraph":
+            realClassName = "Model"
+        elif realClassName == "Input":
+            realClassName = "Layer"
```
Comment (hkvision) [inline pyspark/bigdl/nn/layer.py]: What if the real class is a `KerasModel`? Or in other words, if `Sequential`, how do you identify whether it is a Torch Sequential or a Keras Sequential?

**Comment by zhichao-li** [comment]
jenkins passed  http://172.168.2.101:8080/job/BigDL-PR-Validation-v3/385/

**Pull request merged**